feat: adds waffle flags for legacy libraries v1 and new libraries v2 [FC-0062]

pomegranited opened the pull request:
<!--

Note: Please refer to the Support Development Guidelines on the wiki page to consider backporting to active releases:
https://openedx.atlassian.net/wiki/spaces/COMM/pages/4248436737/Support+Guidelines+for+active+releases

Please give your pull request a short but descriptive title.
Use conventional commits to separate and summarize commits logically:
https://open-edx-proposals.readthedocs.io/en/latest/oep-0051-bp-conventional-commits.html

Use this template as a guide. Omit sections that don't apply.
You may link to information rather than copy it, but only if the link is publicly readable.
If the linked information must be private (because it contains secrets), clearly label the link as private.

-->

## Description

Adds waffle flags to replace the Authoring MFE's `LIBRARY_MODE` env config variable, and exposes them with the contentstore "home" REST API.

Since waffle flags are disabled by default, and we want both v1 and v2 libraries enabled by default in the Authoring MFE, we've named these flags as `disable_*_libraries`.

* `contentstore.new_studio_mfe.disable_legacy_libraries` -> negated to provide `libraries_v1_enabled` in the REST API
* `contentstore.new_studio_mfe.disable_new_libraries` -> negated and combined with `settings.MEILISEARCH_ENABLED` to provide `libraries_v2_enabled` in the REST API.

Combined with https://github.com/openedx/frontend-app-authoring/pull/1329, this change impacts Course Authors.

## Supporting information

Relates to: 
* https://github.com/openedx/frontend-app-authoring/issues/1336
* https://github.com/openedx/frontend-app-authoring/issues/1324
* https://github.com/openedx/edx-platform/issues/32457

Private-ref: [FAL-3858](https://tasks.opencraft.com/browse/FAL-3858)

## Testing instructions

See https://github.com/openedx/frontend-app-authoring/pull/1329

## Deadline

ASAP

## Concerns

1. There is already a `settings.FEATURES` flag called `ENABLE_CONTENT_LIBRARIES`. How this flag should affect use of libraries v1 and/or v2 is [under discussion](https://github.com/openedx/frontend-app-authoring/pull/1329#issuecomment-2389088972).

pomegranited committed:
refactor: rename CourseHomeSerializer to StudioHomeSerializer

to better reflect how this serializer/API is used in the Authoring MFE

pomegranited committed:
feat: adds waffle flags for legacy libraries v1 and new libraries v2 in new Studio Home

These waffle flags replace the MFE env flag LIBRARY_MODE.

openedx-webhooks commented:
Thanks for the pull request, @pomegranited!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

pomegranited committed:
refactor: use contentstore.toggles.libraries_v1_enabled()

and ENABLE_CONTENT_LIBRARIES feature toggle where possible.

ENABLE_CONTENT_LIBRARIES has been incorporated into both the
libraries_v1_enabled() and libraries_v2_enabled() toggles:
FEATURES['ENABLE_CONTENT_LIBRARIES'] must be true for either version of
libraries to be "enabled".

bradenmacdonald reviewed (approved):
Nice work!

I note that `contentstore.new_studio_mfe.disable_new_libraries` doesn't disable the v2 libraries REST APIs, but I think that's good.

I tested various combinations of this, and I did confirm that `contentstore.new_studio_mfe.disable_legacy_libraries` does disable the legacy UI library views like http://studio.local.openedx.io:8001/library/library-v1:foo+bar (now gives 404 if that waffle flag is enabled).

I note that because this depends on Meilisearch, even though v2 libraries will now generally be on by default, they will remain off on edx.org (which is not ready to roll them out yet), because it doesn't have Meilisearch configured. So merging this should not affect 2U.
